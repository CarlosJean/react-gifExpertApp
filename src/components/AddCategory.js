import { useState } from "react";
import { PropTypes } from "prop-types";

export const AddCategory = ({setCategories}) => {

    const [category, setCategory] = useState('');

    const handleSubmit = (e)=>{
        e.preventDefault();        
        if (category.trim().length > 2) {
            setCategories( categories =>[category, ...categories]);
            setCategory('');
        }
    };

    const handleOnChange = (e) => {
        setCategory(e.target.value);         
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{category}</p>
            <input 
                type="text"
                placeholder="Ingrese algo"
                onChange={handleOnChange}
                value={category}
            />
            <button type="reset">Limpiar</button>
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}