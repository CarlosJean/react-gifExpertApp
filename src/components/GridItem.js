import PropTypes from "prop-types";

export const GridItem = ({title, url}) => {
    return (
        <div className="card animate__animated animate__fadeIn">
            <img 
                alt={title}
                src={url}
            />
            <p>{title}</p>
        </div>
    );
}

GridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}