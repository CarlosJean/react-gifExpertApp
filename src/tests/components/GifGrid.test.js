import { shallow } from 'enzyme';

import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs'); //Crea el mock del custom hook.

describe('Pruebas al componente <GifGrid />', () => {

    const category = 'cheesburguer';

    test('El componente debe coincidir con el snapshot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( < GifGrid category = {category}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar items', () => {
        useFetchGifs.mockReturnValue({
            data: [{
                id: '12348946qd6',
                title: 'Cualquier cosa',
                url: 'https://algo.com/cualquier/cosa'
            }],
            loading: false
        });

        const wrapper = shallow( < GifGrid category = {category}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GridItem').exists()).toBe(true);
    });
});