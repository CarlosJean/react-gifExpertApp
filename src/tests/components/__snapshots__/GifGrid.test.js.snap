// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <GifGrid /> Debe mostrar items 1`] = `
<Fragment>
  <h3>
    cheesburguer
  </h3>
  <div
    className="card-grid"
  >
    <GridItem
      id="12348946qd6"
      key="12348946qd6"
      title="Cualquier cosa"
      url="https://algo.com/cualquier/cosa"
    />
  </div>
</Fragment>
`;

exports[`Pruebas al componente <GifGrid /> El componente debe coincidir con el snapshot 1`] = `
<Fragment>
  <h3>
    cheesburguer
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
